alicloud/resource_alicloud_instance.go

```diff
@@ -1160,27 +1161,49 @@ func modifyInstanceType(d *schema.ResourceData, meta interface{}, run bool) (boo
 			return update, err
 		}
 
-		d.SetPartial("instance_type")
-
-		//An instance that was successfully modified once cannot be modified again within 5 minutes.
-		args := ecs.CreateModifyInstanceSpecRequest()
-		args.InstanceId = d.Id()
-		args.InstanceType = d.Get("instance_type").(string)
-		args.ClientToken = buildClientToken(args.GetActionName())
-
-		err = resource.Retry(6*time.Minute, func() *resource.RetryError {
-			_, err = client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
-				return ecsClient.ModifyInstanceSpec(args)
+		instanceChargeType := d.Get("instance_charge_type").(string)
+		switch {
+		case instanceChargeType == string(common.PrePaid):
+			args := ecs.CreateModifyPrepayInstanceSpecRequest()
```

Comment:
I think `case` is needless because of there is only two charge type: PrePaid and PostPaid. Also, PostPaid is default.

Please use `PrePaid` instead of `common.PrePaid`.

Reply:
done